components/Attachment.tsx

```diff
@@ -0,0 +1,444 @@
+   */
+  resizeMode: string
+  /**
+   * The mode of the component, determining where you're using the component (e.g., 'reel', 'profile', or 'cover' (as in cover photo)).
+   */
+  mode: string
+}
+
+const defaultImageUri = 'https://via.placeholder.com/300x200' //TO-DO change this to whatever we decide the default image will be
+
+/**
+ * Attachment is for any area where the user needs to upload files. This includes audio, video for reels, or image files for profile
+ *
+ * ## Usage
+ *
+ *      <Attachment
+ *       width={'100%'}
+ *       height={400}
+ *       resizeMode="contain"
+ *       mode="reel"
+ *       attachment={attachment}
+ *       btnText="Upload Profile"
+ *       setAttachment={setAttachment}
+ *     />
+ *
+ */
+
+const Attachment = ({
+  width = ButtonSize.defaultWidth,
+  height = ButtonSize.defaultHeight,
+  backgroundColor = Colors.white,
+  borderRadius = 0,
+  btnText,
+  attachment,
+  setAttachment,
+  resizeMode = 'contain',
+  mode = 'reel',
+}: Props) => {
+  // Controlling the video async
+  const videoRef = useRef<Video>(null)
+  //for light mode dark mode
+  const colorScheme = useColorScheme()
+  const [audioRef, setAudioRef] = useState<Audio.Sound | null>(null)
+  const [isPlaying, setIsPlaying] = useState<boolean>(false)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+
+  const gradientColors =
+    colorScheme === 'light'
+      ? [Colors.buttonGradientLight.primary, Colors.buttonGradientLight.secondary]
+      : [Colors.buttonGradientDark.primary, Colors.buttonGradientDark.secondary]
+
+  const iconColor = colorScheme === 'light' ? Colors.light.icon : Colors.dark.icon
+  const textColor = colorScheme === 'light' ? Colors.light.text : Colors.dark.text
+
+  // Used just for controlling the play button based on if it's audio or a video
+  const handlePlayPress = async () => {
+    if (attachment.type === 'video') {
+      if (isPlaying) {
+        await videoRef.current.pauseAsync()
+      } else {
+        await videoRef.current.playAsync()
+      }
+    } else if (attachment.type === 'audio') {
+      if (!audioRef.current) {
+        const { sound } = await Audio.Sound.createAsync(
+          { uri: attachment.uri },
+          { shouldPlay: false },
+        )
+        audioRef.current = sound
+      }
+      if (isPlaying) {
+        await audioRef.current.pauseAsync()
+      } else {
+        await audioRef.current.playAsync()
+      }
+    }
+    setIsPlaying(!isPlaying)
+  }
+
+  useEffect(() => {
+    return () => {
+      // When cancelling the audio upload, destroy audio component.
+      if (audioRef && audioRef.current) {
+        audioRef.current.unloadAsync()
+      }
+    }
+  }, [audioRef])
+
+  const selectAttachment = async (
+    setAttachment: React.Dispatch<React.SetStateAction<AttachmentType | null>>,
+    mode: string,
+  ) => {
+    setIsLoading(true)
+
+    let alertOptions = []

```

Comment:
Change to `let alertOptions: any[] = []` to fix the eslint warning.
We could define a custom type for this, but I think its ok not having it as it handles different cases.